File: app/src/main/java/com/ekoapp/simplechat/userlist/UserListAdapter.java
package com.ekoapp.simplechat.userlist;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.ekoapp.ekosdk.EkoObjects;
import com.ekoapp.ekosdk.EkoUser;
import com.ekoapp.ekosdk.adapter.EkoUserAdapter;
import com.ekoapp.simplechat.BaseViewHolder;
import com.ekoapp.simplechat.R;

import butterknife.BindView;

public class UserListAdapter extends EkoUserAdapter<UserListAdapter.UserViewHolder> {

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_user, parent, false);
        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        EkoUser user = getItem(position);

        if (EkoObjects.isProxy(user)) {
            holder.userId = null;
            holder.userTextview.setText("loading...");
        } else {
            String text = new StringBuilder()
                    .append("id: ")
                    .append(user.getUserId())
                    .append("\ndisplayname: ")
                    .append(user.getDisplayName())
                    .append("\nflag count: ")
                    .append(user.getFlagCount())
                    .append("\nmetadata: ")
                    .append(user.getMetadata().toString())
                    .append("\ncreatedAt: ")
                    .append(user.getCreatedAt().toString())
                    .toString();

            holder.userId = user.getUserId();
            holder.userTextview.setText(text);
        }
    }


    static class UserViewHolder extends BaseViewHolder {

        String userId;

        @BindView(R.id.user_textview)
        TextView userTextview;


        UserViewHolder(View itemView) {
            super(itemView);
            itemView.setOnClickListener(view -> {
                // do nothing
            });
        }
    }
}

File: app/src/main/java/com/ekoapp/simplechat/userlist/UserListAdapter.kt
package com.ekoapp.simplechat.userlist

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ekoapp.ekosdk.adapter.EkoUserAdapter
import com.ekoapp.ekosdk.user.EkoUser
import com.ekoapp.simplechat.R
import io.reactivex.BackpressureStrategy
import io.reactivex.Flowable
import io.reactivex.subjects.PublishSubject
import kotlinx.android.synthetic.main.item_user.view.*

class UserListAdapter : EkoUserAdapter<UserListAdapter.UserViewHolder>() {

    private val onLongClickSubject = PublishSubject.create<EkoUser>()
    private val onClickSubject = PublishSubject.create<EkoUser>()

    val onLongClickFlowable: Flowable<EkoUser>
        get() = onLongClickSubject.toFlowable(BackpressureStrategy.BUFFER)

    val onClickFlowable: Flowable<EkoUser>
        get() = onClickSubject.toFlowable(BackpressureStrategy.BUFFER)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_user, parent, false)
        return UserViewHolder(view)
    }

    override fun onBindViewHolder(holder: UserViewHolder, position: Int) {
        val user = getItem(position)
        if (user == null) {
            holder.itemView.user_textview.text = "loading..."
        } else {
            val text = StringBuilder()
                    .append("id: ")
                    .append(user.getUserId())
                    .append("\ndisplayname: ")
                    .append(user.getDisplayName())
                    .append("\nflag count: ")
                    .append(user.getFlagCount())
                    .append("\nmetadata: ")
                    .append(user.getMetadata().toString())
                    .append("\navatarURL: ")
                    .append(user.getAvatar()?.getUrl())
                    .append("\navatarCustomURL: ")
                    .append(user.getAvatarCustomUrl())
                    .append("\ndescription: ")
                    .append(user.getDescription())
                    .append("\ncreatedAt: ")
                    .append(user.getCreatedAt().toString())
                    .toString()
            holder.itemView.user_textview.text = text
            holder.itemView.setOnClickListener {
                onClickSubject.onNext(user)
            }
            holder.itemView.setOnLongClickListener {
                onLongClickSubject.onNext(user)
                true
            }
        }
    }

    class UserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}